docs/pictures/zesong/model.puml

```diff
@@ -0,0 +1,26 @@
+@startuml
```

Comment:
<img width="971" alt="Screenshot 2020-10-29 at 9 19 22 AM" src="https://user-images.githubusercontent.com/46109197/97513687-e2d92500-19c7-11eb-9aa5-76e0d2820bfc.png">

Hi I feel that the sections for component should be more detailed. Instead of describing what a class does, perhaps more explanation of the inner workings of the classes would be helpful.